tck/src/main/java/org/eclipse/microprofile/lra/tck/TckRecoveryTests.java

```diff
@@ -0,0 +1,203 @@
+        // deploy the test service
+        deployer.deploy(DEPLOYMENT_NAME);
+        
+        // invoke phase 1 inside the container (start new LRA and enlist resource)
+        Response response1 = deployementTarget.path(RecoveryResource.RECOVERY_RESOURCE_PATH)
+            .path(RecoveryResource.PHASE_1).request().get();
+        
+        Assert.assertEquals(200, response1.getStatus());
+        URI lra = URI.create(response1.readEntity(String.class));
+
+        // kill the test service while LRA is still active
+        deployer.undeploy(DEPLOYMENT_NAME);
+
+        // start the test service again
+        deployer.deploy(DEPLOYMENT_NAME);
+
+        // invoke phase 2 inside the container (cancel the LRA and verify that the
+        // @Compensate method was called on the enlisted resource)
+        Response response2 = deployementTarget.path(RecoveryResource.RECOVERY_RESOURCE_PATH)
+            .path(RecoveryResource.PHASE_2)
+            .request()
+                .header(RecoveryResource.LRA_HEADER, lra)
+                .get();
+        
+        Assert.assertEquals(response2.readEntity(String.class), 200, response2.getStatus());
+        
+        deployer.undeploy(DEPLOYMENT_NAME);
+    }
+
+    /**
+     * This test verifies that if the microservice application which
+     * enlisted with the LRA fails and the LRA is ended during the time
+     * the service is still down, the Compensate callbacks are received
+     * when the microservice application is started again.
+     * 
+     * Scenario:
+     * - start a new container with a single LRA resource
+     * - start a new LRA and enlist the LRA resource with it
+     * - kill the container
+     * - cancel the LRA while the container is still down
+     * - start the container again
+     * - replay the end phase to get Compensate calls redelivered
+     * - verify that the Compensate callbacks have been received
+     */
+    @Test
+    public void testOneFailureWithCancelDuringRecovery() {
+        // deploy the test service
+        deployer.deploy(DEPLOYMENT_NAME);
+
+        // invoke phase 1 inside the container (start new LRA and enlist resource) which cancels after 500 millis
+        Response response1 = deployementTarget.path(RecoveryResource.RECOVERY_RESOURCE_PATH)
+            .path(RecoveryResource.PHASE_1)
+            .queryParam("timeout", true)
+            .request().get();
+
+        Assert.assertEquals(200, response1.getStatus());
+        URI lra = URI.create(response1.readEntity(String.class));
+
+        // kill the test service while LRA is still active
+        deployer.undeploy(DEPLOYMENT_NAME);
+        
+        // Wait for the timeout cancellation of the LRA. This will put the LRA into cancel only state.
+        // Then wait for the long delay to actually perform the cancellation while the service is still down.
+        // Compensate should be attempted to be called while the participant service is down
```

Comment:
Nice test.

The method name is a bit vague. Shouldn't it be something like testOneFailureWithCancelWhenParticipantIsUnavailable. Also what does OneFailure mean? Do you anticipate a test where there are multiple participants that fail?